src/release_artifacts_resources/ios/cdk/cdk/distribution/github_action_role.py

```diff
@@ -0,0 +1,35 @@
+from aws_cdk import core, aws_iam, aws_s3
+
+class GithubActionRole(core.Construct):
```

Comment:
This `GithubActionRole` construct is dedicated for OIDC with Github Actions. It's not only about the `aws-sdk-ios` package. The OIDC provider created here could be used for any Github Repos.

Reply:
will add the comments

Reply:
i saw this part, so i was wondering whether it's only for one repo? maybe i'm wrong.
`"token.actions.githubusercontent.com:sub": "repo:aws-amplify/aws-sdk-ios:ref:refs/heads/release"`

Reply:
This is because the role that assumed by this principal should only from `aws-sdk-ios` repo. Other repo's workflow should receive 403 if assuming the IAM role.